File: se/se6029/mob-programming/01-guinea-pig-car-car/src/GuineaPig/GuineaPig.java
// GuineaPig/GuineaPig.java

package GuineaPig;

public class GuineaPig {
    protected String name;
    protected int foodCount;

    public GuineaPig() {
        this("GuineaPig");
    }

    public GuineaPig(String name) {
        this.name = name;
        this.foodCount = 0;
    }

    public void noise() {
        System.out.println(this.name + "PuiPui.");
    }

    public void eat(String food) {
        System.out.println(this.name + " eat " + food);
    }

    public void pupu() {
        System.out.println(this.name + " pupu.");
        this.foodCount = 0;
    }

    public String getName() {
        return this.name;
    }
}

File: se/se6029/mob-programming/self-practice/01-guinea-pig-car-car-cpp/src/guinea_pig.cpp
// guinea_pig.cpp

#include "guinea_pig.h"
#include <iostream>

GuineaPig::GuineaPig() : name("GuineaPig"), foodCount(0) {}

GuineaPig::GuineaPig(std::string name) : name(name), foodCount(0) {}

void GuineaPig::noise() {
    std::cout << name << " PuiPui." << std::endl;
}

void GuineaPig::eat(std::string food) {
    std::cout << name << " eat " << food << std::endl;
}

void GuineaPig::pupu() {
    std::cout << name << " pupu." << std::endl;
    foodCount = 0;
}

std::string GuineaPig::getName() {
    return name;
}
